pkg/operator/ceph/object/notification/obc_label_controller.go

```diff
@@ -149,37 +152,60 @@ func (r *ReconcileOBCLabels) reconcile(request reconcile.Request) (reconcile.Res
 				logger.Warningf("bucket notification label mismatch. ignoring key %q value %q", labelKey, labelValue)
 				continue
 			}
-			// for each notification label fetch the bucket notification CRD
+			labelList = append(labelList, labelValue)
 			logger.Debugf("bucket notification label %q found on ObjectbucketClaim %q", labelValue, bucketName)
-			notification := &cephv1.CephBucketNotification{}
-			bnName := types.NamespacedName{Namespace: obc.Namespace, Name: labelValue}
-			if err := r.client.Get(r.opManagerContext, bnName, notification); err != nil {
-				if kerrors.IsNotFound(err) {
-					logger.Infof("CephBucketNotification %q not found", bnName)
-					return waitForRequeueIfNotificationNotReady, nil
-				}
-				return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve CephBucketNotification %q", bnName)
-			}
+		}
+	}
 
-			// get the topic associated with the notification, and make sure it is provisioned
-			topicName := types.NamespacedName{Namespace: notification.Namespace, Name: notification.Spec.Topic}
-			bucketTopic, err := topic.GetProvisioned(r.client, r.opManagerContext, topicName)
-			if err != nil {
-				logger.Infof("CephBucketTopic %q not provisioned yet", topicName)
-				return waitForRequeueIfTopicNotReady, nil
-			}
+	// remove notifications which are no longer specified in the OBC labels
+	for _, oldValue := range bnList {
+		if !sets.NewString(labelList...).Has(oldValue) {
+			deleteList = append(deleteList, oldValue)
+		}
+	}
+	for _, notificationId := range deleteList {
+		err = deleteNotificationFunc(p, &ob, notificationId)
+		if err != nil {
+			logger.Warningf("notification %q failed remove from %q, returned error %v", notificationId, ob.Spec.Endpoint.BucketName, err)
```

Comment:
Is this a warning and not an error? @yuvalif

Reply:
IMO, should be an error

Reply:
Error for the log message or Do we need to return the error as well?

Reply:
I was meaning that it is probably an error that means we should return it and try to reconcile again. Does that still make sense to you @yuvalif ?

Reply:
yes. we should return a retry indication with timeout.
would suggest that we don't bail out in the middle of the loop, to allow other deletes/adds/updates to succeed.
remember that a retry is needed and do that after the loop